### PR TITLE
Fix more warnings from FindBugs

### PR DESCRIPTION
**Pull request by stweil**
Signed-off-by: Stefan Weil <sw@weilnetz.de>

### CODE REVIEW

File: Goobi/src/org/goobi/production/properties/ProcessProperty.java [hunk 1]
```diff
@@ -59,9 +59,12 @@
 	private boolean currentStepDuplicationAllowed = false;
 	
 	public ProcessProperty() {
+		this.container = new Integer(0);
```
Comment (henning-gerhardt) [inline Goobi/src/org/goobi/production/properties/ProcessProperty.java]: IntelliJ IDEA tolds me to remove this unnecessary boxing.

**Comment by henning-gerhardt** [comment]
For commit: https://github.com/kitodo/kitodo-production/pull/507/commits/cb89fd05a3d14c0d91047e35b7caf6c930b1e418
You can not remove this private methods which are used by Hibernate:
```
Could not find a getter for datentyp in class de.sub.goobi.beans.Prozesseigenschaft
```
The other removed methods are maybe used too. So please test your changes in a proper test environment.

File: Goobi/src/org/goobi/production/properties/ProcessProperty.java [hunk 2]
```diff
@@ -291,9 +294,6 @@ public boolean isValid() {
 	 */
 
 	public void save(Schritt step) {
-		if (this.prozesseigenschaft != null) {
-
-		}
```
Comment (henning-gerhardt) [inline Goobi/src/org/goobi/production/properties/ProcessProperty.java]: save method is not used nor defined in interface. Safe to remove?

File: Goobi/src/org/goobi/production/properties/ProcessProperty.java (hunk 1 above)
Comment (stweil) [inline Goobi/src/org/goobi/production/properties/ProcessProperty.java]: I think we have more of that kind, so they can be handled in a new commit.

File: Goobi/src/org/goobi/production/properties/ProcessProperty.java (hunk 2 above)
Comment (stweil) [inline Goobi/src/org/goobi/production/properties/ProcessProperty.java]: Yes, I assume so. I added a new commit which removes it.

**Comment by stweil** [comment]
PR update:
* Removed commit which removed private methods which were reported to be unused but which are obviously used nevertheless.
* Added commit which removes `save` method.

File: Goobi/src/org/goobi/production/properties/ProcessProperty.java (hunk 1 above)
Comment (henning-gerhardt) [inline Goobi/src/org/goobi/production/properties/ProcessProperty.java]: Introducing a known bad usage to change / fix it later is not a good way. We need no more bad usage which must be fixed sometime.

Comment (stweil) [inline Goobi/src/org/goobi/production/properties/ProcessProperty.java]: Sorry, that was a misunderstanding on my side. It's fixed now with an updated PR.

**Comment by stweil** [comment]
New PR, addressed @henning-gerhardt comment.

**Pull request merged**